library/entropy.c

```diff
@@ -363,12 +364,17 @@ int mbedtls_entropy_func( void *data, unsigned char *output, size_t len )
         if( ( ret = entropy_gather_internal( ctx ) ) != 0 )
             goto exit;
 
-        done = 1;
+        thresholds_reached = 1;
+        strong_size = 0;
         for( i = 0; i < ctx->source_count; i++ )
+        {
             if( ctx->source[i].size < ctx->source[i].threshold )
-                done = 0;
+                thresholds_reached = 0;
+            if( ctx->source[i].strong == MBEDTLS_ENTROPY_SOURCE_STRONG )
```

Comment:
Is it possible to reach `ENTROPY_MAX_LOOP`  here if `ctx->source[i].strong == MBEDTLS_ENTROPY_SOURCE_STRONG ` is `false`? There's no way to increase `strong_size` in such case.

Reply:
Yes, but only if there is no strong source. And in this case the call to `entropy_gather_internal` would have returned `MBEDTLS_ERR_ENTROPY_NO_STRONG_SOURCE`.